### PR TITLE
docs(lib/mocha.js): Update API documentation

### PR DESCRIPTION
**Pull request by plroebuck**
### Description of the Change

Replaced all method headers with more complete documentation, better following recommended styles. Eliminated `@api` tags, as `@access` _should_ imply same. Used `@chainable` to denote class methods returning `this` instance.

Renamed a handful of variables, but **no logic changes**!

### Alternate Designs

N/A

### Why should this be in core?

N/A

### Benefits

API documentation will be more useful if more complete.

### Possible Drawbacks

N/A

### Applicable issues

semver-patch

### CODE REVIEW

**Comment by plroebuck** [comment]
No idea how _adding_ documentation could _decrease_ code coverage...

\*\*facepalm\*\*

File: lib/mocha.js [hunk 1]
```diff
@@ -110,11 +121,13 @@ function Mocha(options) {
...
 
 /**
- * Enable or disable bailing on the first failure.
+ * Enables or disables bailing on the first failure.
  *
  * @public
- * @api public
- * @param {boolean} [bail]
+ * @see {@link https://mochajs.org/#-b---bail|CLI option}
+ * @param {boolean} [bail=true] - Whether to bail on first error.
```
Comment (Munter) [inline lib/mocha.js]: Is `bail` really true by default? This doesn't match the default CLI behavior

Comment (plroebuck) [reply lib/mocha.js]: It doesn't?
```js
Mocha.prototype.bail = function(bail) {
  if (!arguments.length) {
    bail = true;
  }
  this.suite.bail(bail);
  return this;
};
```

Comment (plroebuck) [reply lib/mocha.js]: Bear in mind, I was documenting _intent_, not buggy behavior. And yes, there are plenty I found while documenting just this file.

**Pull request merged**